tests/test_kernels.py

```diff
@@ -700,6 +700,74 @@ def test_connectivities_key_kernel(self, adata: AnnData):
         np.testing.assert_array_equal(T_cr.A, adata.obsp[key])
 
 
+class TestVelocityKernelReadData:
```

Comment:
This test is not really necessary, as all the previous tests read from `attr = 'layers'`.
Would rather create 1 new tests where you can truncate velocities from `.layers` and put them in `obsm.`

Reply:
I think it makes sense to keep the warning tbh - and I will need an `elif` or `else` case anyways to set `gene_subset = None`